integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -356,6 +364,212 @@ medusaIntegrationTestRunner({
+                { name: "test location" },
+                adminHeaders
+              )
+            ).data.stock_location
+
+            location2 = (
+              await api.post(
+                `/admin/stock-locations`,
+                { name: "test location 2" },
+                adminHeaders
+              )
+            ).data.stock_location
+
+            salesChannel1 = await createSalesChannel(
+              { name: "sales channel test" },
+              [product.id, product2.id]
+            )
+
+            const api1Res = await api.post(
+              `/admin/api-keys`,
+              { title: "Test publishable KEY", type: ApiKeyType.PUBLISHABLE },
+              adminHeaders
+            )
+
+            publishableKey1 = api1Res.data.api_key
+
+            await api.post(
+              `/admin/api-keys/${publishableKey1.id}/sales-channels`,
+              { add: [salesChannel1.id] },
+              adminHeaders
+            )
+
+            inventoryItem1 = (
+              await api.post(
+                `/admin/inventory-items`,
+                { sku: "test-sku" },
+                adminHeaders
+              )
+            ).data.inventory_item
+
+            inventoryItem2 = (
+              await api.post(
+                `/admin/inventory-items`,
+                { sku: "test-sku-2" },
+                adminHeaders
+              )
+            ).data.inventory_item
+
+            inventoryItem1 = (
+              await api.post(
+                `/admin/inventory-items/${inventoryItem1.id}/location-levels`,
+                {
+                  location_id: location1.id,
+                  stocked_quantity: 20,
+                },
+                adminHeaders
+              )
+            ).data.inventory_item
+
+            inventoryItem2 = (
+              await api.post(
+                `/admin/inventory-items/${inventoryItem2.id}/location-levels`,
+                {
+                  location_id: location2.id,
+                  stocked_quantity: 30,
+                },
+                adminHeaders
+              )
+            ).data.inventory_item
+
+            const remoteLink = appContainer.resolve(
+              ContainerRegistrationKeys.REMOTE_LINK
+            )
+
+            await remoteLink.create([
+              {
+                [Modules.SALES_CHANNEL]: { sales_channel_id: salesChannel1.id },
+                [Modules.STOCK_LOCATION]: { stock_location_id: location1.id },
+              },
+              {
+                [Modules.SALES_CHANNEL]: { sales_channel_id: salesChannel1.id },
+                [Modules.STOCK_LOCATION]: { stock_location_id: location2.id },
+              },
+            ])
+          })
+
+          it("should list all inventory items for a variant", async () => {
+            const remoteLink = appContainer.resolve(
+              ContainerRegistrationKeys.REMOTE_LINK
+            )
+
+            await remoteLink.create([
```

Comment:
Shouldn't this be supported through the product HTTP API? If we are doing this simply because it isn't yet, let's add a TODO here